chatbot.js

```diff
@@ -13,85 +13,93 @@ fs.readFile("./privateRooms.json", "utf8", function (err, data) {
+          } else {
+            privateRooms[user].welcoming = undefined;
+          }
         } else {
-          privateRooms[user].welcoming = undefined;
+          sendInternalMessage("I didn't recognize that response :(", user, client);
         }
-      } else {
-        sendInternalMessage("I didn't recognize that response :(", user, client);
-      }
-    } else if ((!privateRooms[user] || !privateRooms[user].welcoming) && user != client.credentials.userId) {
-      if (privateRooms[user] && privateRooms[user].room == room.roomId) {
-        for (var key in questions) {
-          if (questions.hasOwnProperty(key) && checkForRoomQuestions(msg, key, room.roomId, user, client)) {
-            break;
+      } else if ((!privateRooms[user] || !privateRooms[user].welcoming) && user != client.credentials.userId) {
+        if (privateRooms[user] && privateRooms[user].room == room.roomId) {
+          for (let key in questions) {
+            if (questions.hasOwnProperty(key) && checkForRoomQuestions(msg, key, room.roomId, user, client)) {
+              break;
+            }
           }
+        } else {
+          checkForRoomQuestions(msg, room.roomId, room.roomId, user, client);
         }
-      } else {
-        checkForRoomQuestions(msg, room.roomId, room.roomId, user, client);
       }
-    }
-  } else if (event.getType() === "m.room.member" && event.event.membership === "leave") {
-    var privateRoom = privateRooms[event.getSender()];
-    if (privateRoom && privateRoom.room == event.event.room_id) {
-      privateRoom.room = undefined;
-      privateRoom.welcoming = undefined;
-      savePrivateRooms();
+    } else if (event.getType() === "m.room.member" && event.event.membership === "leave") {
+      let privateRoom = privateRooms[event.getSender()];
+      if (privateRoom && privateRoom.room == event.event.room_id) {
+        privateRoom.room = undefined;
+        privateRoom.welcoming = undefined;
+        savePrivateRooms();
+      }
     }
   }
 };
 
+function checkUser(user) {
+  // Ignore Slack bridge users
+  return !user.startsWith("slack_giveth_");
+}
+
 function checkForRoomQuestions(msg, roomForQuestions, roomToSendIn, user, client) {
-  var questionsForRoom = questions[roomForQuestions];
+  let questionsForRoom = questions[roomForQuestions];
   if (questionsForRoom) {
-    for (var i = 0; i < questionsForRoom.length; i++) {
-      var question = questionsForRoom[i];
-      var shouldAnswerQuestion = false;
+    for (let i = 0; i < questionsForRoom.length; i++) {
+      let question = questionsForRoom[i];
+      let shouldAnswerQuestion = false;
       if (typeof question.trigger === "string") {
         shouldAnswerQuestion = msg.toLowerCase().includes(question.trigger.toLowerCase());
       } else {
-        for (var t = 0; t < question.trigger.length; t++) {
+        for (let t = 0; t < question.trigger.length; t++) {
```

Comment:
This should probably be some foreach cycle to improve readability.

Also I know that the code below is not yours, but comparing two lower case string is kinda weak. It should be some regex ignoring blank spaces and potentially not perfect matches. I guess this is more feedback for improvement :)

Reply:
I agree with using forEach - should be done to all for loops in this case. Although a rather minor detail with using regex, which really doesn't matter in this case tbh